File: sdk/node/examples/stream_pipeline.mjs
// Stream producer → consumer demo: send three byte chunks through a
// Message.StreamHandle and drain them on the other side.
import { Stream } from "../reflow.mjs";

const s = Stream.create({
  bufferSize: 16,
  originActor: "producer",
  originPort: "out",
  contentType: "application/octet-stream",
});

for (const buf of [Buffer.from("alpha"), Buffer.from("bravo"), Buffer.from("charlie")]) {
  s.sendBytes(buf);
}
s.end();

const msg = s.intoMessage();
console.log("produced:", msg.kind);

const rx = msg.takeStream();
const collected = [];
let done = false;
while (!done) {
  const f = await rx.recv(500);
  switch (f.kind) {
    case "begin":
      console.log("begin", f.contentType);
      break;
    case "data":
      collected.push(Buffer.from(f.data).toString());
      break;
    case "end":
    case "closed":
      done = true;
      break;
    case "error":
      console.error("stream error:", f.error);
      done = true;
      break;
    case "timeout":
      done = true;
      break;
  }
}
console.log("collected:", collected);

File: sdk/python/examples/stream_pipeline.py
"""Stream producer → consumer: send three byte chunks through a
Message.StreamHandle and drain them on the other side."""

from offbit_reflow import Stream


def main() -> None:
    s = Stream.create(
        buffer_size=16,
        origin_actor="producer",
        origin_port="out",
        content_type="application/octet-stream",
    )
    for buf in (b"alpha", b"bravo", b"charlie"):
        s.send_bytes(buf)
    s.end()

    msg = s.into_message()
    print("produced:", msg.kind)

    rx = msg.take_stream()
    collected = []
    while True:
        f = rx.recv(500)
        kind = f["kind"]
        if kind == "data":
            collected.append(bytes(f["data"]))
        elif kind in ("end", "closed"):
            break
        elif kind == "error":
            raise RuntimeError(f"stream error: {f.get('error')}")
        elif kind == "timeout":
            break
    print("collected:", collected)


if __name__ == "__main__":
    main()
